dynamic-config/config-consul/src/main/java/org/apache/servicecomb/config/consul/ConsulDynamicPropertiesSource.java

```diff
@@ -0,0 +1,157 @@
+    consulConfigProperties.setDelayTime(consulConfig.getConsulDelayTime());
+    return consulConfigProperties;
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(value = ConsulRawClient.Builder.class, parameterizedContainer = Supplier.class)
+  public Supplier<ConsulRawClient.Builder> consulRawClientBuilderSupplier() {
+    return createConsulRawClientBuilder();
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  public ConsulClient consulClient(ConsulConfigProperties consulConfigProperties,
+                                   Supplier<ConsulRawClient.Builder> consulRawClientBuilderSupplier) {
+    return createConsulClient(consulConfigProperties, consulRawClientBuilderSupplier);
+  }
+
+  public static Supplier<ConsulRawClient.Builder> createConsulRawClientBuilder() {
+    return ConsulRawClient.Builder::builder;
+  }
+
+  public static ConsulClient createConsulClient(ConsulConfigProperties consulConfigProperties,
+                                                Supplier<ConsulRawClient.Builder> consulRawClientBuilderSupplier) {
+    ConsulRawClient.Builder builder = consulRawClientBuilderSupplier.get();
+    final String agentHost = StringUtils.hasLength(consulConfigProperties.getScheme())
+        ? consulConfigProperties.getScheme() + "://" + consulConfigProperties.getHost() : consulConfigProperties.getHost();
+    builder.setHost(agentHost).setPort(consulConfigProperties.getPort());
+
+    return new ConsulClient(builder.build());
+  }
+
+  @Bean(name = CONFIG_WATCH_TASK_SCHEDULER_NAME)
+  @ConditionalOnMissingBean
+  public TaskScheduler configWatchTaskScheduler() {
+    ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
+    threadPoolTaskScheduler.setPoolSize(8);
+    threadPoolTaskScheduler.setRemoveOnCancelPolicy(true);
+    threadPoolTaskScheduler.setThreadNamePrefix("consul-config-");
+
+    threadPoolTaskScheduler.initialize();
+    return threadPoolTaskScheduler;
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  public ConsulConfigClient consulConfigClient(Environment environment) {
+    ConsulConfigProperties consulConfigProperties = consulConfigProperties(environment);
+    ConsulClient consulClient = consulClient(consulConfigProperties, consulRawClientBuilderSupplier());
+    UpdateHandler updateHandler = updateHandler();
+    TaskScheduler taskScheduler = configWatchTaskScheduler();
+    return new ConsulConfigClient(updateHandler, environment, consulConfigProperties, consulClient, taskScheduler);
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  @ConditionalOnBean(ConsulConfigClient.class)
+  public ConsulConfigClientClear consulConfigClear() {
+    return new ConsulConfigClientClear();
+  }
+
+  @Override
+  public PropertySource<?> create(Environment environment) {
+    try {
+      consulConfigClient = consulConfigClient(environment);
+      ConsulConfigClientClear consulConfigClientClear = consulConfigClear();
+      LOGGER.info("consulConfigClientClear getPhase:{}", consulConfigClientClear.getPhase());
```

Comment:
`ConsulConfigClientClear` can be deleted? Seems no use. And this code slip will create many `ConsulConfigClient` instance and  is not correct